Crash when running two instances of the same lsp?

### Summary

I launched two instances of helix for different projects, within 5 seconds-ish of each other, however, the second instance crashed when I launched the lsp (I used :lsp-restart).

Here is a video after it apparently crashed:

https://github.com/helix-editor/helix/assets/79347623/c255209b-a890-4a2f-8401-2c9da8076028



### Reproduction Steps

<!-- Ideally provide a key sequence and/or asciinema.org recording. --> 

I tried this:

With the repos https://github.com/haskell/haskell-language-server and https://github.com/kowainik/stan
1. In one terminal session (stan) `hx **/**Analysis.hs`, in the other (haskell-language-server) `hx **/**Stan.hs`
2. :lsp-restart in the haskell-language-server one
I expected this to happen:
Load the two projects independently without problems.

Instead, this happened:
Helix crashed.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-12-16T17:30:09.829 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:30:09.829003Z | Info | Live bytes: 324.02MB Heap size: 1093.66MB\n"
2023-12-16T17:30:19.737 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:30:19.736792Z | Info | Live bytes: 332.86MB Heap size: 968.88MB\n"
2023-12-16T17:34:32.969 helix_lsp::transport [ERROR] haskell-language-server err <- "No 'hie.yaml' found. Try to discover the project type!\n"
2023-12-16T17:34:32.976 helix_lsp::transport [ERROR] haskell-language-server err <- "Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper) Version 2.5.0.0 aarch64 ghc-9.2.8\n"
2023-12-16T17:34:32.976 helix_lsp::transport [ERROR] haskell-language-server err <- "Current directory: /Users/roger/Documents/Dev/haskell-proj/stan\n"
2023-12-16T17:34:32.976 helix_lsp::transport [ERROR] haskell-language-server err <- "Operating system: darwin\n"
2023-12-16T17:34:32.977 helix_lsp::transport [ERROR] haskell-language-server err <- "Arguments: [\"--lsp\"]\n"
2023-12-16T17:34:32.977 helix_lsp::transport [ERROR] haskell-language-server err <- "Cradle directory: /Users/roger/Documents/Dev/haskell-proj/stan\n"
2023-12-16T17:34:32.977 helix_lsp::transport [ERROR] haskell-language-server err <- "Cradle type: Cabal\n"
2023-12-16T17:34:33.187 helix_lsp::transport [ERROR] haskell-language-server err <- "\n"
2023-12-16T17:34:33.187 helix_lsp::transport [ERROR] haskell-language-server err <- "Tool versions found on the $PATH\n"
2023-12-16T17:34:33.188 helix_lsp::transport [ERROR] haskell-language-server err <- "cabal:          3.10.2.0\n"
2023-12-16T17:34:33.188 helix_lsp::transport [ERROR] haskell-language-server err <- "stack:          2.13.1\n"
2023-12-16T17:34:33.188 helix_lsp::transport [ERROR] haskell-language-server err <- "ghc:            9.6.3\n"
2023-12-16T17:34:33.188 helix_lsp::transport [ERROR] haskell-language-server err <- "\n"
2023-12-16T17:34:33.188 helix_lsp::transport [ERROR] haskell-language-server err <- "\n"
2023-12-16T17:34:33.188 helix_lsp::transport [ERROR] haskell-language-server err <- "Consulting the cradle to get project GHC version...\n"
2023-12-16T17:34:33.194 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:33.190632Z | Debug | executing command: cabal exec -v0 -- ghc --print-libdir\n"
2023-12-16T17:34:33.741 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:33.741220Z | Debug | executing command: cabal exec -v0 -- ghc -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:34.636 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:34.635316Z | Debug | executing command: cabal --builddir=/Users/roger/.cache/hie-bios/dist-stan-e7fdc7a1a68a772ee2734d2160941891 v2-exec --with-compiler /Users/roger/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36 --with-hc-pkg /Users/roger/.cache/hie-bios/ghc-pkg-27a5b43f9dd36e7642d843de9b490023 ghc -v0 -- --numeric-version\n"
2023-12-16T17:34:35.083 helix_lsp::transport [ERROR] haskell-language-server err <- "Project GHC version: 9.6.3\n"
2023-12-16T17:34:35.083 helix_lsp::transport [ERROR] haskell-language-server err <- "haskell-language-server exe candidates: [\"haskell-language-server-9.6.3\",\"haskell-language-server\"]\n"
2023-12-16T17:34:35.085 helix_lsp::transport [ERROR] haskell-language-server err <- "Launching haskell-language-server exe at:/Users/roger/.ghcup/bin/haskell-language-server-9.6.3\n"
2023-12-16T17:34:35.085 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:35.085605Z | Debug | executing command: cabal exec -v0 -- ghc --print-libdir\n"
2023-12-16T17:34:35.218 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:35.217928Z | Debug | executing command: cabal exec -v0 -- ghc -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:35.502 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:35.501339Z | Debug | executing command: cabal --builddir=/Users/roger/.cache/hie-bios/dist-stan-e7fdc7a1a68a772ee2734d2160941891 v2-exec --with-compiler /Users/roger/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36 --with-hc-pkg /Users/roger/.cache/hie-bios/ghc-pkg-27a5b43f9dd36e7642d843de9b490023 ghc -v0 -- -v0 -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:35.784 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:35.784568Z | Debug | executing command: cabal exec -v0 -- ghc --print-libdir\n"
2023-12-16T17:34:35.918 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:35.917848Z | Debug | executing command: cabal exec -v0 -- ghc -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:36.202 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:36.201270Z | Debug | executing command: cabal --builddir=/Users/roger/.cache/hie-bios/dist-stan-e7fdc7a1a68a772ee2734d2160941891 v2-exec --with-compiler /Users/roger/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36 --with-hc-pkg /Users/roger/.cache/hie-bios/ghc-pkg-27a5b43f9dd36e7642d843de9b490023 ghc -v0 -- --print-libdir\n"
2023-12-16T17:34:38.142 helix_lsp::transport [ERROR] haskell-language-server err <- "Found \"/Users/roger/Documents/Dev/haskell-proj/haskell-language-server/hie.yaml\" for \"/Users/roger/Documents/Dev/haskell-proj/haskell-language-server/a\"\n"
2023-12-16T17:34:38.147 helix_lsp::transport [ERROR] haskell-language-server err <- "Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper) Version 2.5.0.0 aarch64 ghc-9.2.8\n"
2023-12-16T17:34:38.147 helix_lsp::transport [ERROR] haskell-language-server err <- "Current directory: /Users/roger/Documents/Dev/haskell-proj/haskell-language-server\n"
2023-12-16T17:34:38.148 helix_lsp::transport [ERROR] haskell-language-server err <- "Operating system: darwin\n"
2023-12-16T17:34:38.148 helix_lsp::transport [ERROR] haskell-language-server err <- "Arguments: [\"--lsp\"]\n"
2023-12-16T17:34:38.148 helix_lsp::transport [ERROR] haskell-language-server err <- "Cradle directory: /Users/roger/Documents/Dev/haskell-proj/haskell-language-server\n"
2023-12-16T17:34:38.148 helix_lsp::transport [ERROR] haskell-language-server err <- "Cradle type: Cabal\n"
2023-12-16T17:34:38.241 helix_lsp::transport [ERROR] haskell-language-server err <- "\n"
2023-12-16T17:34:38.241 helix_lsp::transport [ERROR] haskell-language-server err <- "Tool versions found on the $PATH\n"
2023-12-16T17:34:38.241 helix_lsp::transport [ERROR] haskell-language-server err <- "cabal:          3.10.2.0\n"
2023-12-16T17:34:38.241 helix_lsp::transport [ERROR] haskell-language-server err <- "stack:          2.13.1\n"
2023-12-16T17:34:38.241 helix_lsp::transport [ERROR] haskell-language-server err <- "ghc:            9.6.3\n"
2023-12-16T17:34:38.241 helix_lsp::transport [ERROR] haskell-language-server err <- "\n"
2023-12-16T17:34:38.241 helix_lsp::transport [ERROR] haskell-language-server err <- "\n"
2023-12-16T17:34:38.241 helix_lsp::transport [ERROR] haskell-language-server err <- "Consulting the cradle to get project GHC version...\n"
2023-12-16T17:34:38.242 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:38.241817Z | Debug | executing command: cabal exec -v0 -- ghc --print-libdir\n"
2023-12-16T17:34:38.465 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:38.465178Z | Info | haskell-language-server version: 2.5.0.0 (GHC: 9.6.3) (PATH: /Users/roger/.ghcup/hls/2.5.0.0/lib/haskell-language-server-2.5.0.0/bin/haskell-language-server-9.6.3)\n"
2023-12-16T17:34:38.466 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:38.466286Z | Info | Directory: /Users/roger/Documents/Dev/haskell-proj/stan\n"
2023-12-16T17:34:38.468 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:38.466476Z | Info | Starting (haskell-language-server) LSP server...\n"
2023-12-16T17:34:38.468 helix_lsp::transport [ERROR] haskell-language-server err <- "  GhcideArguments {argsCommand = LSP, argsCwd = Nothing, argsShakeProfiling = Nothing, argsTesting = False, argsExamplePlugin = False, argsLogLevel = Info, argsLogFile = Nothing, argsLogStderr = True, argsLogClient = False, argsThreads = 0, argsProjectGhcVersion = False}\n"
2023-12-16T17:34:38.468 helix_lsp::transport [ERROR] haskell-language-server err <- "  PluginIds: [ pragmas-suggest\n"
2023-12-16T17:34:38.468 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-completion\n"
2023-12-16T17:34:38.468 helix_lsp::transport [ERROR] haskell-language-server err <- "             , retrie\n"
2023-12-16T17:34:38.468 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-completions\n"
2023-12-16T17:34:38.468 helix_lsp::transport [ERROR] haskell-language-server err <- "             , alternateNumberFormat\n"
2023-12-16T17:34:38.468 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-bindings\n"
2023-12-16T17:34:38.468 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fields\n"
2023-12-16T17:34:38.468 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-hover-and-symbols\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-type-signatures\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , fourmolu\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-extend-import-action\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-fill-holes\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , importLens\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , LSPRecorderCallback\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , qualifyImportedNames\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , moduleName\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , splice\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stylish-haskell\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , changeTypeSignature\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , hlint\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , class\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ormolu\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , callHierarchy\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stan\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-type-lenses\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , codeRange\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal-fmt\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , eval\n"
2023-12-16T17:34:38.469 helix_lsp::transport [ERROR] haskell-language-server err <- "             , rename\n"
2023-12-16T17:34:38.470 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-imports-exports\n"
2023-12-16T17:34:38.470 helix_lsp::transport [ERROR] haskell-language-server err <- "             , gadt\n"
2023-12-16T17:34:38.470 helix_lsp::transport [ERROR] haskell-language-server err <- "             , overloaded-record-dot\n"
2023-12-16T17:34:38.470 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-core\n"
2023-12-16T17:34:38.470 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fixity\n"
2023-12-16T17:34:38.470 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-disable ]\n"
2023-12-16T17:34:38.470 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:38.470307Z | Info | Logging heap statistics every 60.00s\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- " 2023-12-16T23:34:38.525760Z | Info | Starting LSP server...\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "  If you are seeing this in a terminal, you probably should have run WITHOUT the --lsp option!\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "  PluginIds: [ pragmas-suggest\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-completion\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , retrie\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-completions\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , alternateNumberFormat\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-bindings\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fields\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-hover-and-symbols\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-type-signatures\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , fourmolu\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-extend-import-action\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-fill-holes\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , importLens\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , LSPRecorderCallback\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , qualifyImportedNames\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , moduleName\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , splice\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stylish-haskell\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , changeTypeSignature\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , hlint\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , class\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ormolu\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , callHierarchy\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stan\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-type-lenses\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , codeRange\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal-fmt\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , eval\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , rename\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-imports-exports\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , gadt\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , overloaded-record-dot\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-core\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fixity\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-disable ]\n"
2023-12-16T17:34:38.526 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:38.526641Z | Info | Starting server\n"
2023-12-16T17:34:38.542 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:38.542227Z | Info | Started LSP server in 0.02s\n"
2023-12-16T17:34:39.133 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:39.133278Z | Debug | executing command: cabal exec -v0 -- ghc -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:39.162 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:39.161989Z | Info | Registering IDE configuration: IdeConfiguration {workspaceFolders = fromList [NormalizedUri 7602845792364802616 \"file:///Users/roger/Documents/Dev/haskell-proj/stan\"], clientSettings = hashed (Just (Object (fromList [(\"haskell\",Object (fromList [(\"formattingProvider\",String \"ormolu\")]))])))}\n"
2023-12-16T17:34:39.208 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:39.208088Z | Info | Cradle path: src/Stan/Analysis.hs\n"
2023-12-16T17:34:39.208 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:39.208207Z | Warning | No [cradle](https://github.com/mpickering/hie-bios#hie-bios) found for src/Stan/Analysis.hs.\n"
2023-12-16T17:34:39.208 helix_lsp::transport [ERROR] haskell-language-server err <- "Proceeding with [implicit cradle](https://hackage.haskell.org/package/implicit-hie).\n"
2023-12-16T17:34:39.208 helix_lsp::transport [ERROR] haskell-language-server err <- "You should ignore this message, unless you see a 'Multi Cradle: No prefixes matched' error.\n"
2023-12-16T17:34:39.211 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:39.211209Z | Info | invoking build tool to determine build flags (this may take some time depending on the cache)\n"
2023-12-16T17:34:39.661 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:39.660243Z | Debug | executing command: cabal --builddir=/Users/roger/.cache/hie-bios/dist-haskell-language-server-8d18fbae90afeeb2ac5fd40060ea0c19 v2-exec --with-compiler /Users/roger/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36 --with-hc-pkg /Users/roger/.cache/hie-bios/ghc-pkg-27a5b43f9dd36e7642d843de9b490023 ghc -v0 -- --numeric-version\n"
2023-12-16T17:34:40.486 helix_lsp::transport [ERROR] haskell-language-server err <- "Project GHC version: 9.6.3\n"
2023-12-16T17:34:40.487 helix_lsp::transport [ERROR] haskell-language-server err <- "haskell-language-server exe candidates: [\"haskell-language-server-9.6.3\",\"haskell-language-server\"]\n"
2023-12-16T17:34:40.487 helix_lsp::transport [ERROR] haskell-language-server err <- "Launching haskell-language-server exe at:/Users/roger/.ghcup/bin/haskell-language-server-9.6.3\n"
2023-12-16T17:34:40.487 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:40.487659Z | Debug | executing command: cabal exec -v0 -- ghc --print-libdir\n"
2023-12-16T17:34:40.525 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:40.525200Z | Info | Interface files cache directory: /Users/roger/.cache/ghcide/stan-0.1.1.0-inplace-e6159ec741e3bef7505a45e8e3ed004e4db3196d\n"
2023-12-16T17:34:40.525 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:40.525281Z | Info | Making new HscEnv. In-place unit ids: [stan-0.1.1.0-inplace]\n"
2023-12-16T17:34:40.587 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:40.587066Z | Info | Cradle path: test/Test/Stan/Analysis.hs\n"
2023-12-16T17:34:40.587 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:40.587174Z | Warning | No [cradle](https://github.com/mpickering/hie-bios#hie-bios) found for test/Test/Stan/Analysis.hs.\n"
2023-12-16T17:34:40.587 helix_lsp::transport [ERROR] haskell-language-server err <- "Proceeding with [implicit cradle](https://hackage.haskell.org/package/implicit-hie).\n"
2023-12-16T17:34:40.587 helix_lsp::transport [ERROR] haskell-language-server err <- "You should ignore this message, unless you see a 'Multi Cradle: No prefixes matched' error.\n"
2023-12-16T17:34:40.589 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:40.589534Z | Info | invoking build tool to determine build flags (this may take some time depending on the cache)\n"
2023-12-16T17:34:40.806 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:40.806419Z | Debug | executing command: cabal exec -v0 -- ghc -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:41.458 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:41.456666Z | Debug | executing command: cabal --builddir=/Users/roger/.cache/hie-bios/dist-haskell-language-server-8d18fbae90afeeb2ac5fd40060ea0c19 v2-exec --with-compiler /Users/roger/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36 --with-hc-pkg /Users/roger/.cache/hie-bios/ghc-pkg-27a5b43f9dd36e7642d843de9b490023 ghc -v0 -- -v0 -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:42.646 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:42.646540Z | Debug | executing command: cabal exec -v0 -- ghc --print-libdir\n"
2023-12-16T17:34:43.089 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:43.088476Z | Debug | executing command: cabal exec -v0 -- ghc -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:44.149 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:44.147824Z | Debug | executing command: cabal --builddir=/Users/roger/.cache/hie-bios/dist-haskell-language-server-8d18fbae90afeeb2ac5fd40060ea0c19 v2-exec --with-compiler /Users/roger/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36 --with-hc-pkg /Users/roger/.cache/hie-bios/ghc-pkg-27a5b43f9dd36e7642d843de9b490023 ghc -v0 -- --print-libdir\n"
2023-12-16T17:34:47.239 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:47.237738Z | Info | haskell-language-server version: 2.5.0.0 (GHC: 9.6.3) (PATH: /Users/roger/.ghcup/hls/2.5.0.0/lib/haskell-language-server-2.5.0.0/bin/haskell-language-server-9.6.3)\n"
2023-12-16T17:34:47.240 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:47.238994Z | Info | Directory: /Users/roger/Documents/Dev/haskell-proj/haskell-language-server\n"
2023-12-16T17:34:47.240 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:47.239991Z | Info | Starting (haskell-language-server) LSP server...\n"
2023-12-16T17:34:47.241 helix_lsp::transport [ERROR] haskell-language-server err <- "  GhcideArguments {argsCommand = LSP, argsCwd = Nothing, argsShakeProfiling = Nothing, argsTesting = False, argsExamplePlugin = False, argsLogLevel = Info, argsLogFile = Nothing, argsLogStderr = True, argsLogClient = False, argsThreads = 0, argsProjectGhcVersion = False}\n"
2023-12-16T17:34:47.241 helix_lsp::transport [ERROR] haskell-language-server err <- "  PluginIds: [ pragmas-suggest\n"
2023-12-16T17:34:47.241 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-completion\n"
2023-12-16T17:34:47.241 helix_lsp::transport [ERROR] haskell-language-server err <- "             , retrie\n"
2023-12-16T17:34:47.241 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-completions\n"
2023-12-16T17:34:47.241 helix_lsp::transport [ERROR] haskell-language-server err <- "             , alternateNumberFormat\n"
2023-12-16T17:34:47.241 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-bindings\n"
2023-12-16T17:34:47.241 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fields\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-hover-and-symbols\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-type-signatures\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , fourmolu\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-extend-import-action\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-fill-holes\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , importLens\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , LSPRecorderCallback\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , qualifyImportedNames\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , moduleName\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , splice\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stylish-haskell\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , changeTypeSignature\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , hlint\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , class\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ormolu\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , callHierarchy\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stan\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-type-lenses\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , codeRange\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal-fmt\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , eval\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , rename\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-imports-exports\n"
2023-12-16T17:34:47.242 helix_lsp::transport [ERROR] haskell-language-server err <- "             , gadt\n"
2023-12-16T17:34:47.243 helix_lsp::transport [ERROR] haskell-language-server err <- "             , overloaded-record-dot\n"
2023-12-16T17:34:47.243 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-core\n"
2023-12-16T17:34:47.243 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fixity\n"
2023-12-16T17:34:47.243 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-disable ]\n"
2023-12-16T17:34:47.243 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:47.243333Z | Info | Logging heap statistics every 60.00s\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- " 2023-12-16T23:34:47.353088Z | Info | Starting LSP server...\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "  If you are seeing this in a terminal, you probably should have run WITHOUT the --lsp option!\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "  PluginIds: [ pragmas-suggest\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-completion\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , retrie\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-completions\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , alternateNumberFormat\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-bindings\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fields\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-hover-and-symbols\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-type-signatures\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , fourmolu\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-extend-import-action\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-fill-holes\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , importLens\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , LSPRecorderCallback\n"
2023-12-16T17:34:47.354 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , qualifyImportedNames\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , moduleName\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , splice\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stylish-haskell\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , changeTypeSignature\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , hlint\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , class\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ormolu\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , callHierarchy\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stan\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-type-lenses\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , codeRange\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal-fmt\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , eval\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , rename\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-imports-exports\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , gadt\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , overloaded-record-dot\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-core\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fixity\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-disable ]\n"
2023-12-16T17:34:47.355 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:47.354633Z | Info | Starting server\n"
2023-12-16T17:34:47.360 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:47.359653Z | Info | Started LSP server in 0.01s\n"
2023-12-16T17:34:49.235 helix_lsp::client [WARN] language server failed to terminate gracefully - server closed the stream
2023-12-16T17:34:49.289 helix_lsp::transport [ERROR] haskell-language-server err <- "Found \"/Users/roger/Documents/Dev/haskell-proj/haskell-language-server/hie.yaml\" for \"/Users/roger/Documents/Dev/haskell-proj/haskell-language-server/a\"\n"
2023-12-16T17:34:49.289 helix_lsp::transport [ERROR] haskell-language-server err <- "Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper) Version 2.5.0.0 aarch64 ghc-9.2.8\n"
2023-12-16T17:34:49.291 helix_lsp::transport [ERROR] haskell-language-server err <- "Current directory: /Users/roger/Documents/Dev/haskell-proj/haskell-language-server\n"
2023-12-16T17:34:49.291 helix_lsp::transport [ERROR] haskell-language-server err <- "Operating system: darwin\n"
2023-12-16T17:34:49.291 helix_lsp::transport [ERROR] haskell-language-server err <- "Arguments: [\"--lsp\"]\n"
2023-12-16T17:34:49.291 helix_lsp::transport [ERROR] haskell-language-server err <- "Cradle directory: /Users/roger/Documents/Dev/haskell-proj/haskell-language-server\n"
2023-12-16T17:34:49.291 helix_lsp::transport [ERROR] haskell-language-server err <- "Cradle type: Cabal\n"
2023-12-16T17:34:49.516 helix_lsp::transport [ERROR] haskell-language-server err <- "\n"
2023-12-16T17:34:49.516 helix_lsp::transport [ERROR] haskell-language-server err <- "Tool versions found on the $PATH\n"
2023-12-16T17:34:49.516 helix_lsp::transport [ERROR] haskell-language-server err <- "cabal:          3.10.2.0\n"
2023-12-16T17:34:49.516 helix_lsp::transport [ERROR] haskell-language-server err <- "stack:          2.13.1\n"
2023-12-16T17:34:49.517 helix_lsp::transport [ERROR] haskell-language-server err <- "ghc:            9.6.3\n"
2023-12-16T17:34:49.517 helix_lsp::transport [ERROR] haskell-language-server err <- "\n"
2023-12-16T17:34:49.517 helix_lsp::transport [ERROR] haskell-language-server err <- "\n"
2023-12-16T17:34:49.517 helix_lsp::transport [ERROR] haskell-language-server err <- "Consulting the cradle to get project GHC version...\n"
2023-12-16T17:34:49.518 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:49.517968Z | Debug | executing command: cabal exec -v0 -- ghc --print-libdir\n"
2023-12-16T17:34:50.186 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:50.185163Z | Debug | executing command: cabal exec -v0 -- ghc -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:50.246 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:50.246306Z | Info | Registering IDE configuration: IdeConfiguration {workspaceFolders = fromList [NormalizedUri (-5314531846057779252) \"file:///Users/roger/Documents/Dev/haskell-proj/haskell-language-server\"], clientSettings = hashed (Just (Object (fromList [(\"haskell\",Object (fromList [(\"formattingProvider\",String \"ormolu\")]))])))}\n"
2023-12-16T17:34:50.247 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:50.246710Z | Error | Got EOF\n"
2023-12-16T17:34:50.260 helix_lsp::transport [ERROR] haskell-language-server err: <- StreamClosed
2023-12-16T17:34:51.297 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:51.295085Z | Debug | executing command: cabal --builddir=/Users/roger/.cache/hie-bios/dist-haskell-language-server-8d18fbae90afeeb2ac5fd40060ea0c19 v2-exec --with-compiler /Users/roger/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36 --with-hc-pkg /Users/roger/.cache/hie-bios/ghc-pkg-27a5b43f9dd36e7642d843de9b490023 ghc -v0 -- --numeric-version\n"
2023-12-16T17:34:52.043 helix_lsp::transport [ERROR] haskell-language-server err <- "Project GHC version: 9.6.3\n"
2023-12-16T17:34:52.043 helix_lsp::transport [ERROR] haskell-language-server err <- "haskell-language-server exe candidates: [\"haskell-language-server-9.6.3\",\"haskell-language-server\"]\n"
2023-12-16T17:34:52.045 helix_lsp::transport [ERROR] haskell-language-server err <- "Launching haskell-language-server exe at:/Users/roger/.ghcup/bin/haskell-language-server-9.6.3\n"
2023-12-16T17:34:52.045 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:52.045288Z | Debug | executing command: cabal exec -v0 -- ghc --print-libdir\n"
2023-12-16T17:34:52.728 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:52.723921Z | Debug | executing command: cabal exec -v0 -- ghc -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:53.809 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:53.806895Z | Debug | executing command: cabal --builddir=/Users/roger/.cache/hie-bios/dist-haskell-language-server-8d18fbae90afeeb2ac5fd40060ea0c19 v2-exec --with-compiler /Users/roger/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36 --with-hc-pkg /Users/roger/.cache/hie-bios/ghc-pkg-27a5b43f9dd36e7642d843de9b490023 ghc -v0 -- -v0 -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:54.894 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:54.892848Z | Debug | executing command: cabal exec -v0 -- ghc --print-libdir\n"
2023-12-16T17:34:55.577 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:55.575002Z | Debug | executing command: cabal exec -v0 -- ghc -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)\n"
2023-12-16T17:34:56.695 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:34:56.694096Z | Debug | executing command: cabal --builddir=/Users/roger/.cache/hie-bios/dist-haskell-language-server-8d18fbae90afeeb2ac5fd40060ea0c19 v2-exec --with-compiler /Users/roger/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36 --with-hc-pkg /Users/roger/.cache/hie-bios/ghc-pkg-27a5b43f9dd36e7642d843de9b490023 ghc -v0 -- --print-libdir\n"
2023-12-16T17:35:00.033 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:00.032226Z | Info | haskell-language-server version: 2.5.0.0 (GHC: 9.6.3) (PATH: /Users/roger/.ghcup/hls/2.5.0.0/lib/haskell-language-server-2.5.0.0/bin/haskell-language-server-9.6.3)\n"
2023-12-16T17:35:00.033 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:00.033305Z | Info | Directory: /Users/roger/Documents/Dev/haskell-proj/haskell-language-server\n"
2023-12-16T17:35:00.033 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:00.033485Z | Info | Starting (haskell-language-server) LSP server...\n"
2023-12-16T17:35:00.034 helix_lsp::transport [ERROR] haskell-language-server err <- "  GhcideArguments {argsCommand = LSP, argsCwd = Nothing, argsShakeProfiling = Nothing, argsTesting = False, argsExamplePlugin = False, argsLogLevel = Info, argsLogFile = Nothing, argsLogStderr = True, argsLogClient = False, argsThreads = 0, argsProjectGhcVersion = False}\n"
2023-12-16T17:35:00.034 helix_lsp::transport [ERROR] haskell-language-server err <- "  PluginIds: [ pragmas-suggest\n"
2023-12-16T17:35:00.035 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-completion\n"
2023-12-16T17:35:00.035 helix_lsp::transport [ERROR] haskell-language-server err <- "             , retrie\n"
2023-12-16T17:35:00.035 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-completions\n"
2023-12-16T17:35:00.035 helix_lsp::transport [ERROR] haskell-language-server err <- "             , alternateNumberFormat\n"
2023-12-16T17:35:00.035 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-bindings\n"
2023-12-16T17:35:00.035 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fields\n"
2023-12-16T17:35:00.035 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-hover-and-symbols\n"
2023-12-16T17:35:00.036 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-type-signatures\n"
2023-12-16T17:35:00.036 helix_lsp::transport [ERROR] haskell-language-server err <- "             , fourmolu\n"
2023-12-16T17:35:00.036 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-extend-import-action\n"
2023-12-16T17:35:00.036 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-fill-holes\n"
2023-12-16T17:35:00.036 helix_lsp::transport [ERROR] haskell-language-server err <- "             , importLens\n"
2023-12-16T17:35:00.036 helix_lsp::transport [ERROR] haskell-language-server err <- "             , LSPRecorderCallback\n"
2023-12-16T17:35:00.036 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , qualifyImportedNames\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , moduleName\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , splice\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stylish-haskell\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , changeTypeSignature\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , hlint\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , class\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ormolu\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , callHierarchy\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stan\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-type-lenses\n"
2023-12-16T17:35:00.037 helix_lsp::transport [ERROR] haskell-language-server err <- "             , codeRange\n"
2023-12-16T17:35:00.038 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal-fmt\n"
2023-12-16T17:35:00.038 helix_lsp::transport [ERROR] haskell-language-server err <- "             , eval\n"
2023-12-16T17:35:00.038 helix_lsp::transport [ERROR] haskell-language-server err <- "             , rename\n"
2023-12-16T17:35:00.038 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-imports-exports\n"
2023-12-16T17:35:00.038 helix_lsp::transport [ERROR] haskell-language-server err <- "             , gadt\n"
2023-12-16T17:35:00.038 helix_lsp::transport [ERROR] haskell-language-server err <- "             , overloaded-record-dot\n"
2023-12-16T17:35:00.038 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-core\n"
2023-12-16T17:35:00.038 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fixity\n"
2023-12-16T17:35:00.038 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-disable ]\n"
2023-12-16T17:35:00.039 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:00.038911Z | Info | Logging heap statistics every 60.00s\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- " 2023-12-16T23:35:00.104971Z | Info | Starting LSP server...\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "  If you are seeing this in a terminal, you probably should have run WITHOUT the --lsp option!\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "  PluginIds: [ pragmas-suggest\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-completion\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , retrie\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-completions\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , alternateNumberFormat\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-bindings\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fields\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-hover-and-symbols\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-type-signatures\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , fourmolu\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-extend-import-action\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-fill-holes\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , importLens\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , LSPRecorderCallback\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , qualifyImportedNames\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , moduleName\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , splice\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stylish-haskell\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , changeTypeSignature\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , hlint\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , class\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ormolu\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , callHierarchy\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , stan\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-type-lenses\n"
2023-12-16T17:35:00.105 helix_lsp::transport [ERROR] haskell-language-server err <- "             , codeRange\n"
2023-12-16T17:35:00.106 helix_lsp::transport [ERROR] haskell-language-server err <- "             , cabal-fmt\n"
2023-12-16T17:35:00.106 helix_lsp::transport [ERROR] haskell-language-server err <- "             , eval\n"
2023-12-16T17:35:00.106 helix_lsp::transport [ERROR] haskell-language-server err <- "             , rename\n"
2023-12-16T17:35:00.106 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-code-actions-imports-exports\n"
2023-12-16T17:35:00.106 helix_lsp::transport [ERROR] haskell-language-server err <- "             , gadt\n"
2023-12-16T17:35:00.106 helix_lsp::transport [ERROR] haskell-language-server err <- "             , overloaded-record-dot\n"
2023-12-16T17:35:00.106 helix_lsp::transport [ERROR] haskell-language-server err <- "             , ghcide-core\n"
2023-12-16T17:35:00.106 helix_lsp::transport [ERROR] haskell-language-server err <- "             , explicit-fixity\n"
2023-12-16T17:35:00.106 helix_lsp::transport [ERROR] haskell-language-server err <- "             , pragmas-disable ]\n"
2023-12-16T17:35:00.106 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:00.105858Z | Info | Starting server\n"
2023-12-16T17:35:00.107 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:00.107304Z | Info | Started LSP server in 0.00s\n"
2023-12-16T17:35:02.074 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:02.074573Z | Info | Registering IDE configuration: IdeConfiguration {workspaceFolders = fromList [NormalizedUri (-5314531846057779252) \"file:///Users/roger/Documents/Dev/haskell-proj/haskell-language-server\"], clientSettings = hashed (Just (Object (fromList [(\"haskell\",Object (fromList [(\"formattingProvider\",String \"ormolu\")]))])))}\n"
2023-12-16T17:35:02.123 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:02.122744Z | Info | Cradle path: plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs\n"
2023-12-16T17:35:02.123 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:02.123118Z | Info | invoking build tool to determine build flags (this may take some time depending on the cache)\n"
2023-12-16T17:35:05.779 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:05.779037Z | Info | Interface files cache directory: /Users/roger/.cache/ghcide/hls-stan-plugin-2.5.0.0-inplace-0b70b4a795fd8552d423567d4726688d91e610b5\n"
2023-12-16T17:35:05.779 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:05.779121Z | Info | Making new HscEnv. In-place unit ids: [hls-stan-plugin-2.5.0.0-inplace]\n"
2023-12-16T17:35:34.020 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:34.020579Z | Info | Interface files cache directory: /Users/roger/.cache/ghcide/main-a2bbfd2e9c1e97155908acdc80717588d28e2f6d-a2bbfd2e9c1e97155908acdc80717588d28e2f6d\n"
2023-12-16T17:35:34.021 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:34.020799Z | Info | Interface files cache directory: /Users/roger/.cache/ghcide/stan-0.1.1.0-inplace-a2bbfd2e9c1e97155908acdc80717588d28e2f6d\n"
2023-12-16T17:35:34.021 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:34.020818Z | Info | Making new HscEnv. In-place unit ids: [ main-a2bbfd2e9c1e97155908acdc80717588d28e2f6d\n"
2023-12-16T17:35:34.021 helix_lsp::transport [ERROR] haskell-language-server err <- "                                      , stan-0.1.1.0-inplace ]\n"
2023-12-16T17:35:38.503 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:35:38.503678Z | Info | Live bytes: 217.88MB Heap size: 1001.39MB\n"
2023-12-16T17:35:52.007 helix_lsp::transport [ERROR] haskell-language-server err: <- StreamClosed
2023-12-16T17:35:52.007 helix_lsp::client [WARN] language server failed to terminate gracefully - server closed the stream
2023-12-16T17:35:52.008 helix_lsp::transport [ERROR] haskell-language-server err: <- IO(Os { code: 32, kind: BrokenPipe, message: "Broken pipe" })
2023-12-16T17:36:38.505 helix_lsp::transport [ERROR] haskell-language-server err <- "2023-12-16T23:36:38.504903Z | Info | Live bytes: 217.88MB Heap size: 1001.39MB\n"

```

</details>


### Platform

macOS

### Terminal Emulator

kitty 0.31.0 

### Installation Method

source

### Helix Version

23.05-492-g31f50bf5

--- a/helix-lsp/src/lib.rs
+++ b/helix-lsp/src/lib.rs
@@ -915,10 +915,17 @@ fn start_client(
         }
 
         // next up, notify<initialized>
-        _client
+        let notification_result = _client
             .notify::<lsp::notification::Initialized>(lsp::InitializedParams {})
-            .await
-            .unwrap();
+            .await;
+
+        if let Err(e) = notification_result {
+            log::error!(
+                "failed to notify language server of its initialization: {}",
+                e
+            );
+            return;
+        }
 
         initialize_notify.notify_one();
     });

